test/es-module/test-require-module-preload.js

```diff
@@ -6,65 +6,98 @@ const fixtures = require('../common/fixtures');
 
 const stderr = /ExperimentalWarning: Support for loading ES Module in require/;
 
-// Test named exports.
-{
-  spawnSyncAndExitWithoutError(
-    process.execPath,
-    [ '--experimental-require-module', '-r', fixtures.path('../fixtures/es-module-loaders/module-named-exports.mjs') ],
-    {
-      stderr,
-    }
-  );
-}
+function testPreload(preloadFlag) {
+  // Test named exports.
+  {
+    spawnSyncAndExitWithoutError(
+      process.execPath,
+      [
+        '--experimental-require-module',
+        preloadFlag,
+        fixtures.path('es-module-loaders/module-named-exports.mjs'),
+        fixtures.path('printA.js'),
+      ],
+      {
+        stdout: 'A',
+        stderr,
+        trim: true,
+      }
+    );
+  }
 
-// Test ESM that import ESM.
-{
-  spawnSyncAndExitWithoutError(
-    process.execPath,
-    [ '--experimental-require-module', '-r', fixtures.path('../fixtures/es-modules/import-esm.mjs') ],
-    {
-      stderr,
-      stdout: 'world',
-      trim: true,
-    }
-  );
-}
+  // Test ESM that import ESM.
+  {
+    spawnSyncAndExitWithoutError(
+      process.execPath,
+      [
+        '--experimental-require-module',
+        preloadFlag,
+        fixtures.path('es-modules/import-esm.mjs'),
```

Comment:
This is not correct, `--import` should accept URLs, not paths.

Reply:
It actually takes absolute paths, just that on Windows, `c:` paths would be recognized as URLs...I wonder if that's something that we want to fix, though that's unrelated to this PR. I'll just use the URL in the test for now for Windows.

Reply:
It does not accept path, only URLs AFAICT:

```console
$ echo 'console.log("URL") > '/tmp/path with space'
$ echo 'console.log("path")' > '/tmp/path%20with%20spaces'
$ ./node --import /tmp/path%20with%20spaces test/fixtures/empty.js
URL
$ echo 'console.log("URL") > '/tmp/path'
$ echo 'console.log("path")' > '/tmp/path?test'
$ ./node --import '/tmp/path?test' test/fixtures/empty.js
URL
$ ./node --require '/tmp/path?test' test/fixtures/empty.js
path
```

It's just that Unix paths can be accepted as relative URLs – but that breaks as soon as there's any URL-special char.

Can we use a relative path/URL instead? So if the repo is cloned / the tarball is extracted on a path with such char, the test doesn't fail:
```suggestion
        path.relative(__dirname, fixtures.path('es-modules/import-esm.mjs')).replaceAll('\\', '/'),
```

Reply:
Updated to simply run the process from the fixtures directory, that should be simple enough.